### PR TITLE
Use hypershift binary

### PR DESCRIPTION
**Pull request by ianzhang366**
On my e2e test, it seems the hypershift adds new CRD... which requires the latest version.
As we know, the latest version is conflicting with the `library-go`...

This PR is using the hypershift binary to install and uninstalled the hypershift operator.

https://github.com/stolostron/backlog/issues/20511

### CODE REVIEW

**Comment by ianzhang366** [comment]
@zhiweiyin318 somehow the pre-delete job is not triggered, can you please help me double check if the pre-job is configured correctly.

File: pkg/manager/manifests/templates/pre-delete-job-hypershift.yaml [hunk 1]
```diff
@@ -21,8 +21,6 @@ spec:
       - name: {{ .AddonName }}
         image: {{ .Image }}
         imagePullPolicy: {{ .Values.global.imagePullPolicy }}
```
Comment (zhujian7) [inline pkg/manager/manifests/templates/pre-delete-job-hypershift.yaml:23]: please check the `.Values.global.imagePullPolicy` and `.Values.addonInstallNamespace`, I am not sure we can get it.

**Comment by ianzhang366** [comment]
@zhujian7 can you please review this again?

File: Makefile [hunk 2]
```diff
@@ -62,15 +62,15 @@ vendor:
 	go mod vendor
 
 .PHONY: build
-build: vendor fmt vet ## Build manager binary.
+build: vendor fmt vet build-hypershift ## Build manager binary.
```
Comment (zhujian7) [inline Makefile:65]: Please make sure the DownStream build can support this.

File: go.mod [hunk 3]
```diff
@@ -2,10 +2,16 @@ module github.com/stolostron/hypershift-addon-operator
 
 go 1.17
 
+replace (
+	github.com/stolostron/hypershift-addon-operator/pkg/agent => ./pkg/agent
```
Comment (zhujian7) [inline go.mod:6]: why do we need this?

File: Makefile [hunk 4]
```diff
@@ -80,7 +80,7 @@ docker-push: ## Push docker image with the manager.
 
 .PHONY: build-hypershift
 build-hypershift: ## Push docker image with the manager.
-	build/build-hypershift.sh
+	TargetHypershiftRelease="main" build/build-hypershift.sh
```
[the remark this answers is not in the document]
Comment (ianzhang366) [reply Makefile:83]: you are right, but not sure should we do `release-4.10` or `release-ocm2.5`(which should be the acm branch)

File: Makefile (hunk 2 above)
Comment (ianzhang366) [reply Makefile:65]: eh... have you worked with downstream for this kind of issue... I don't know what to do for downstream...

File: Makefile (hunk 4 above)
[the remark this answers is not in the document]
Comment (ianzhang366) [reply Makefile:83]: this is what I had in mind.. not sure if it's still true

[the remark this answers is not in the document]
Comment (zhujian7) [reply Makefile:83]: Yes, I just found out that hypershift-deployment-controller seems to be using main: https://github.com/stolostron/hypershift-deployment-controller/blob/main/go.mod#L8

[the remark this answers is not in the document]
Comment (ianzhang366) [reply Makefile:83]: https://github.com/stolostron/backlog/issues/19416#issuecomment-1054271702

**Pull request merged**